haystack/components/agents/agent.py

```diff
@@ -238,6 +249,192 @@ def _create_agent_span(self) -> Any:
+
+        if isinstance(snapshot.break_point.break_point, ToolBreakpoint):
+            messages = current_inputs["tool_invoker"]["messages"]
+            skip_chat_generator = True
+        else:
+            messages = current_inputs["chat_generator"]["messages"]
+            skip_chat_generator = False
+
+        state.set("messages", messages)
+
+        streaming_callback = current_inputs["chat_generator"].get("streaming_callback", streaming_callback)
+        streaming_callback = select_streaming_callback(  # type: ignore[call-overload]
+            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=requires_async
+        )
+
+        tool_invoker_inputs: dict[str, Any] = {}
+        generator_inputs: dict[str, Any] = {"tools": self.tools}
+        if streaming_callback is not None:
+            tool_invoker_inputs["streaming_callback"] = streaming_callback
+            generator_inputs["streaming_callback"] = streaming_callback
+
+        return _ExecutionContext(
+            state=state,
+            component_visits=component_visits,
+            chat_generator_inputs=generator_inputs,
+            tool_invoker_inputs=tool_invoker_inputs,
+            counter=snapshot.break_point.break_point.visit_count,
+            skip_chat_generator=skip_chat_generator,
+        )
+
+    def _runtime_checks(self, break_point: Optional[AgentBreakpoint], snapshot: Optional[AgentSnapshot]) -> None:
+        """
+        Perform runtime checks before running the agent.
+
+        :param break_point: An AgentBreakpoint, can be a Breakpoint for the "chat_generator" or a ToolBreakpoint
+            for "tool_invoker".
+        :param snapshot: An AgentSnapshot containing the state of a previously saved agent execution.
+        :raises RuntimeError: If the Agent component wasn't warmed up before calling `run()`.
+        :raises ValueError: If both break_point and snapshot are provided, or if the break_point is invalid.
+        """
+        if not self._is_warmed_up and hasattr(self.chat_generator, "warm_up"):
+            raise RuntimeError("The component Agent wasn't warmed up. Run 'warm_up()' before calling 'run()'.")
+
+        if break_point and snapshot:
+            raise ValueError(
+                "break_point and snapshot cannot be provided at the same time. The agent run will be aborted."
+            )
+
+        if break_point and isinstance(break_point.break_point, ToolBreakpoint):
+            _validate_tool_breakpoint_is_valid(agent_breakpoint=break_point, tools=self.tools)
+
+    def _check_chat_generator_breakpoint(
+        self,
+        execution_context: _ExecutionContext,
+        break_point: Optional[AgentBreakpoint],
+        parent_snapshot: Optional[PipelineSnapshot],
+    ) -> None:
+        """
+        Check if the chat generator breakpoint is triggered.
+
+        If the breakpoint is triggered, create an agent snapshot and check the chat generator breakpoint.
```

Comment:
nit: In the following method, we use handle. Let's use the same word.

Reply:
Could you clarify here? I've updated the underneath method to be `_trigger_chat_generator_breakpoint`. I think `check` still makes sense here since it's not guaranteed that the break point will be triggered.